code/modules/mob/mob_grab.dm

```diff
@@ -317,6 +320,32 @@
 
 	return 1
 
+// Function to compute the current slowdown and is more adjustable and uses mob/movement_delay() as starting value
+// The code will adjust or lower the slowdown depending on STAT_ROB skill, gravity, etc.
+/obj/item/weapon/grab/proc/update_slowdown()
+	// The movment speed of assailant will be determined by the victim whatever their size or things he wears minus how strong the assailant ( ROB )
+	// New function should take the victim variables in account : size of mob, under gravity or not
+	// ROB check will start in process for the victim so the assailant can have a jump on the victim in first movement tick or some shit unless he's already grabbed
+	var/affecting_stat = affecting.stats.getStat(STAT_ROB)	// Victim
+	var/assailant_stat = assailant.stats.getStat(STAT_ROB)	// Grabber
+
+	// initial value for slowdown
+	slowdown = affecting.movement_delay()
```

Comment:
I don't think we should use this one - this takes into account movement intent, damage, inventory and stuff - it's more trouble than it's worth. Just the gravity and size checks should be fine.

Reply:
Actually the type is simply mob, not mob/carbon/human, so it will find his way to code\modules\mob\mob.dm at line 140

Reply:
Correction that's /mob/living/carbon/human/movement_delay()